tox.ini

```diff
@@ -31,11 +31,11 @@ deps =
   wheel
 commands =
   # safety measure to assure we do not accidentaly run tests with broken dependencies
+  # We add coverage options but not making them mandatory as we do not want to force
+  # pytest users to run coverage when they just want to run a single test with `pytest -k test`
   python -m pip check
   {envpython} -m pytest \
-  --cov "{envsitepackagesdir}/ansiblelint" \
-  --junitxml "{toxworkdir}/junit.{envname}.xml" \
-  {posargs:}
+  {posargs:--cov "{envsitepackagesdir}/ansiblelint" --junitxml "{toxworkdir}/junit.{envname}.xml" --cov-report term-missing:skip-covered --cov-report xml:.test-results/pytest/cov.xml --no-cov-on-fail}
```

Comment:
Do tox substitutions work across lines? Multi-line would read better

Reply:
Also, for CI runs I'd rather not set no-cov-on-fail...

Reply:
Also, for CI runs I'd rather not set no-cov-on-fail...

Reply:
With current Github Actions the impact is not very big because sadly github displays output with top instead of tail (opposed to most other CI systems which display the tail and allow you to scroll down, more like a terminal approach).

Please note that removing `--no-cov-on-fail` option would be a huge problem for local development, because it would force any developer running tox to scroll few screens to see the test failed.

IMHO, there is zero value in producing a coverage report when the outcome if fail. Nobody cares about coverage when the outcome is already a failure. Also that report would clearly not be trustable either because it would not report a real coverage.

As recommended by other pytest users, many do have a different `tox -e coverage` environment that does this. CI would use it, and if we do that it could be run all the time, even consolidating data from multiple environments.